### PR TITLE
[WIP] Basic VirtIO support

### PR DESCRIPTION
**Pull request by otyshchenko1**
The VirtIO feature (which this PR adds) allows us to run either DomU or DomA on a VirtIO block device instead of using a traditional Xen PV block device (this PR also introduces a recipe for a completely new entity which is the virtio-disk backend).
This PR was successfully tested on a system with DomU and DomA.
It is a [WIP] because:
- This support means to be disabled by default for now. So the last patch should be dropped (I left it in current PR just for
  a visibility reason)
- I used ioreq_4.14_v5 branch at my Xen repo so far since the product is based on v4.14 release while my IOREQ patch series
  on v4.15-rc. So either the product should be migrated to v4.15-rc (preferred) or a series should be back-ported to release.

### CODE REVIEW

**Comment by otyshchenko1** [comment]
Please note this patch for Android Linux is needed as well:
https://github.com/xen-troops/linux/pull/96

**Comment by otyshchenko1** [comment]
Update Xen and backend branches

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/doc/local.conf.dom0-image-minimal-initramfs [hunk 1]
```diff
@@ -225,6 +225,9 @@ DISTRO_FEATURES_append = " xen"
 # Configuration for Para-virtual camera
 DISTRO_FEATURES_append = " pvcamera"
 
+# Configuration for VirtIO feature
+DISTRO_FEATURES_append = " virtio"
```
Comment (iusyk) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/doc/local.conf.dom0-image-minimal-initramfs:229]: maybe it is reasonable to make an override for the machines to support the ''H3 ES3" ?

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/doma.bb [hunk 2]
```diff
@@ -56,4 +56,19 @@ do_install() {
         # Update GUEST_DEPENDENCIES by adding camerabe after sndbe
         sed -i 's/\<sndbe\>/& camerabe/' ${D}${sysconfdir}/init.d/guest_doma
     fi
+
```
Comment (iusyk) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/doma.bb:59]: here as well, if support the override - we just need to implement the additional command like do_install_append_h3es3

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/domd.bb [hunk 3]
```diff
@@ -60,4 +60,10 @@ do_install() {
 
     install -d ${D}${sysconfdir}/init.d
     install -m 0744 ${WORKDIR}/guest_domd ${D}${sysconfdir}/init.d/
+
+    if ${@bb.utils.contains('DISTRO_FEATURES', 'virtio', 'true', 'false', d)}; then
```
Comment (iusyk) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/domd.bb:64]: the same like above

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/domu.bb [hunk 4]
```diff
@@ -64,4 +64,21 @@ do_install() {
         # Update GUEST_DEPENDENCIES by adding camerabe after sndbe
         sed -i 's/\<sndbe\>/& camerabe/' ${D}${sysconfdir}/init.d/guest_domu
     fi
+
+    if ${@bb.utils.contains('DISTRO_FEATURES', 'virtio', 'true', 'false', d)}; then
```
Comment (iusyk) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/domu.bb:68]: like for doma

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/doma-vdevices.cfg [hunk 5]
```diff
@@ -6,6 +6,8 @@ vgsx = [ 'backend=DomD, osid=1' ]
 
 # Android uses disk with it's own partitions so no index after xvda
 disk = [ 'backend=DomD, phy:/var/run/android-disks/doma, xvda' ]
+#vdisk = [ 'backend=DomD, disks=rw:/var/run/android-disks/doma' ]
```
Comment (iusyk) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/doma-vdevices.cfg:9]: as i understand, for the case of virtio support, we need to uncomment the code, probably new config like doma-vdevices-virtio.cfg is better solution ?

**Review by iusyk — COMMENTED** [review]
common comment is : consider the overrides instead of 'if' implementation

**Pull request closed without merging**